contracts/infrastructure/DappRegistry.sol

```diff
@@ -7,72 +7,193 @@ import "./storage/Storage.sol";
         return false;
     }
 
-    function toggle(address _wallet, bytes32 _registry) external override onlyModule(_wallet) returns (bool) {
-        require(_registry == ARGENT_DEFAULT || registryManager[_registry] != address(0), "AR: unknow registry");
-        bool current = isEnabled[_wallet][_registry];
-        isEnabled[_wallet][_registry] = !current;
-        return !current;
+    function toggleRegistry(address _wallet, uint8 _registryId, bool _enabled) external override onlyModule(_wallet) returns (bool) {
+        require(_registryId == 0 /* Argent Default Registry */ || registryManagers[_registryId] != address(0), "AR: unknow registry");
+        uint registries = uint(enabledRegistryIds[_wallet]);
+        bool current = ((registries >> _registryId) & 1) > 0;
+        require(current != _enabled, "AR: bad state change" );
+        enabledRegistryIds[_wallet] = bytes32(registries ^ (uint(1) << _registryId)); // toggle [_registryId]^th bit
+    }
+
+    /**************  Management of registry list  *****************/
+
+    function createRegistry(uint8 _registryId, address _manager) external onlyOwner {
+        require(_registryId > 0 && _manager != address(0), "AR: invalid parameters");
+        require(registryManagers[_registryId] == address(0), "AR: duplicate registry");
+        registryManagers[_registryId] = _manager;
+        emit RegistryCreated(_registryId, _manager);
+    }
+
+    // Note: removeRegistry is not supported because that would allow the owner to replace registries that 
+    // have already been enabled by users with a new (potentially maliciously populated) registry 
+
+    function changeManager(uint8 _registryId, address _newManager) external {
+        address manager = registryManagers[_registryId];
+        require(manager != address(0), "AR: unknow registry");
+        require(_newManager != address(0), "AR: new manager is 0");
+        require(msg.sender == manager, "AR: sender should be manager");
+        registryManagers[_registryId] = _newManager;
+    }
+
+    function requestTimelockChange(uint64 _newSecurityPeriod) external onlyOwner {
+        newSecurityPeriod = _newSecurityPeriod;
+        securityPeriodChangeAfter = uint64(block.timestamp) + securityPeriod;
+        emit TimelockChangePended(_newSecurityPeriod);
     }
 
-    // Do we want to let the owner to delete a registry?
-    function createRegistry(bytes32 _registry, address _manager) external onlyOwner {
-        require(_registry != bytes32(0) && _manager != address(0), "AR: invalid parameters");
-        require(registryManager[_registry] == address(0), "AR: duplicate registry");
-        registries.push(_registry);
-        registryManager[_registry] = _manager;
-        emit RegistryCreated(_registry, _manager);
+    function confirmTimelockChange() external {
+        uint64 newPeriod = newSecurityPeriod;
+        require(newPeriod > 0 && securityPeriodChangeAfter <= block.timestamp, "AR: can't (yet) change timelock");
```

Comment:
This check was meant to only allow calls to this method (which can be called by anyone) when a timelock change is pending. I wanted to allow changing the timelock to 0 if needed, so I should have used `securityPeriodChangeAfter` instead of `newPeriod` here:
```
require(securityPeriodChangeAfter > 0 && securityPeriodChangeAfter <= block.timestamp, "AR: can't (yet) change timelock");
```